pkg/pipeline/utils/multiorderedmap_test.go

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMultiOrderedMap_AddRecord(t *testing.T) {
+	mom := NewMultiOrderedMap()
+
+	err := mom.AddRecord(5, "aaaa")
+	require.NoError(t, err)
+	require.Equal(t, 1, mom.Len())
+
+	err = mom.AddRecord(5, "bb")
+	require.Error(t, err)
+	require.Equal(t, 1, mom.Len())
+
+	err = mom.AddRecord(6, "bb")
+	require.NoError(t, err)
+	require.Equal(t, 2, mom.Len())
+	assertLengthConsistency(t, mom)
+}
+
+func TestMultiOrderedMap_GetRecord(t *testing.T) {
+	mom := NewMultiOrderedMap()
+	record, found := mom.GetRecord(5)
+	require.False(t, found)
+	require.Nil(t, record)
+
+	mustAddRecord(t, mom, 5, "aaaa")
+	mustAddRecord(t, mom, 6, "bb")
+
+	record, found = mom.GetRecord(5)
+	require.True(t, found)
+	require.Equal(t, "aaaa", record)
+	record, found = mom.GetRecord(6)
+	require.True(t, found)
+	require.Equal(t, "bb", record)
+	assertLengthConsistency(t, mom)
+}
+
+func TestMultiOrderedMap_RemoveRecord(t *testing.T) {
+	mom := NewMultiOrderedMap()
+	// Test removing non-existing key
+	mom.RemoveRecord(5)
+
+	mustAddRecord(t, mom, 5, "aaaa")
+
+	// Test removing existing key
+	mom.RemoveRecord(5)
+	require.Zero(t, mom.Len())
+	assertLengthConsistency(t, mom)
+}
+
+func TestMultiOrderedMap_MoveToBack(t *testing.T) {
+	lengthOrder := OrderID("length")
+	lexicalOrder := OrderID("lexical")
+	mom := NewMultiOrderedMap(lengthOrder, lexicalOrder)
+
+	mustAddRecord(t, mom, 5, "aaaa")
+	mustAddRecord(t, mom, 6, "ddd")
+	mustAddRecord(t, mom, 7, "b")
+	mustAddRecord(t, mom, 8, "cc")
+
+	err := mom.MoveToBack(7, lengthOrder)
+	require.NoError(t, err)
+	err = mom.MoveToBack(6, lexicalOrder)
+	require.NoError(t, err)
+
+	assertOrder(t, mom, lengthOrder, []string{"aaaa", "ddd", "cc", "b"})
+	assertOrder(t, mom, lexicalOrder, []string{"aaaa", "b", "cc", "ddd"})
+
+	// Test invalid args
+	err = mom.MoveToBack(0, lengthOrder)
+	require.Error(t, err)
+	err = mom.MoveToBack(5, "ERROR")
+	require.Error(t, err)
+	assertLengthConsistency(t, mom)
+}
+
+func TestMultiOrderedMap_IterateFrontToBackIterateFrontToBack(t *testing.T) {
```

Comment:
Is the name IterateFrontToBack doubled intentionally or is it a copy-paste error?

Reply:
it's an error. good catch!